M2/Macaulay2/m2/orderedmonoidrings.m2

```diff
@@ -268,7 +268,21 @@ Ring OrderedMonoid := PolynomialRing => (			  -- no memoize
 		   f=f*RM_(-minexps); -- get rid of monomial in factor if f Laurent polynomial
 		   c=RM_minexps;
 		   );
-	       (facs,exps) := rawFactor raw f;	-- example value: ((11, x+1, x-1, 2x+3), (1, 1, 1, 1)); constant term is first, if there is one
+	       isNumberField := F -> isField F and instance(baseRing F, QuotientRing) and coefficientRing baseRing F === QQ and numgens baseRing F == 1 and numgens ideal baseRing F == 1; 
+	       (facs,exps) := if isNumberField R then (
+        	   baseR := baseRing R;
+		   RM' := QQ(monoid(generators RM|generators baseR)); -- need to put everyone inside the same ring
```

Comment:
Isn't this more easily done with `flattenRing`?

Reply:
I'm afraid not... `flattenRing` treats the field as a field and does not expand it.
```macaulay2
i12 : first flattenRing(toField(QQ[i]/(i^2+1))[x,y])

       QQ[i]
o12 = ------[][x..y]
       2
      i  + 1

o12 : PolynomialRing
```
Also we need the generator of the field to be the last variable, so in this way we can be sure.

Reply:
Try it this way:
```m2
Macaulay2, version 1.17.2.1
with packages: ConwayPolynomials, Elimination, IntegralClosure, InverseSystems, LLLBases,
               MinimalPrimes, PrimaryDecomposition, ReesAlgebra, Saturation, TangentCone

i1 : K = toField ( QQ[x]/(x^2-3) )

o1 = K

o1 : PolynomialRing

i2 : R = K[y]

o2 = R

o2 : PolynomialRing

i6 : flattenRing (R, CoefficientRing => QQ)

      QQ[y, x]       QQ[y, x]
o6 = (--------, map (--------, R, {y, x}))
        2              2
       x  - 3         x  - 3

o6 : Sequence
```

Reply:
Yes you are right! Thanks!